util/settings/settings.go

```diff
@@ -1258,6 +1272,72 @@ func (mgr *SettingsManager) InitializeSettings(insecureModeEnabled bool) (*ArgoC
 	return cdSettings, nil
 }
 
+// Update data with K8S Secret referenced content
+func (mgr *SettingsManager) updateMapSecretRef(data map[string]interface{}) (map[string]interface{}, error) {
+	updatedData := make(map[string]interface{})
+	suffix := "SecretRef"
+	for key, value := range data {
+		if strings.HasSuffix(key, suffix) { // ex. ClientSecretSecretRef
+			var secretRef apiv1.SecretKeySelector
+			// Make sure value apiv1.SecretKeySelector struct
+			tmp, _ := yaml.Marshal(value)
+			err := yaml.Unmarshal(tmp, &secretRef)
+			if err != nil {
+				return data, err
+			}
+
+			// Fetch K8S Secret
+			k8sSecret, err := mgr.clientset.CoreV1().Secrets(mgr.namespace).Get(context.Background(), secretRef.Name, metav1.GetOptions{})
```

Comment:
Argo CD executes `getConfigMap` method very frequently. It is too expensive to make K8S API request. Please use `mgr.GetSecretsLister` instead:

```golang
lister, err := mgr.GetSecretsLister()
if err != nil {
  return nil
}
lister.Secrets(mgr.namespace).Get(secretRef.Name)
```

As a side effect, each referenced secret must have `"app.kubernetes.io/part-of": "argocd"` label but I cannot see any other way to avoid performance degradation.

Reply:
I think we can live with it. Even if secrets are not withing `argocd-secret` K8S secret, they are still _part-of_ argocd. I'll update my code (and tests) accordingly.

Reply:
After further testing, looks like `"app.kubernetes.io/part-of": "argocd"` only applies to ArgoCD `ConfigMap`, meaning `lister.Secrets` is able to look for K8S `Secret` even if they are not label with `"app.kubernetes.io/part-of": "argocd"` . But I might be wrong.